File: springboot-web/json-web-examples/src/main/java/techtabu/json/mapkey/MapKeyController.java
package techtabu.json.mapkey;

import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Map;

/**
 * @author TechTabu
 */

@RestController
@RequestMapping("/mapkey")
@Slf4j
@CrossOrigin("*")
public class MapKeyController {

    @JsonSerialize(keyUsing = PersonKeyWithJSerializer.class)
    Map<PersonKeyWithJ, Person> map;

    @GetMapping("/")
    public Map<PersonKey, Person> getAllPersons() {

        return Map.of(new PersonKey("Tabu", "Dev"), new Person("Tabu", "tabu@email.com", 45000),
                new PersonKey("Neel", "Tabu"), new Person("Neel", "neel@email.com", 43000));
    }

    @GetMapping("/byserializer")
    public Map<PersonKeyWithJ, Person> getAllPersonsByPersonKeyWithJ() {


        map = Map.of(new PersonKeyWithJ("Shalu", "Tabu"), new Person("Shalu", "shalu@email.com", 45000),
                new PersonKeyWithJ("Inara", "Tabu"), new Person("Inara", "inara@email.com", 43000));
        return map;
    }
}

File: springboot-web/json-web-examples/src/main/java/techtabu/json/mapkey/MapKeyController.kt
package techtabu.json.mapkey

import com.fasterxml.jackson.databind.annotation.JsonSerialize
import org.springframework.web.bind.annotation.CrossOrigin
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

/**
 * @author TechTabu
 */
@RestController
@RequestMapping("/mapkey")
@CrossOrigin("*")
class MapKeyController {
    @JsonSerialize(keyUsing = PersonKeyWithJSerializer::class)
    var map: Map<PersonKeyWithJ, Person>? = null

    @get:GetMapping("/")
    val allPersons: Map<PersonKey, Person>
        get() = mapOf(
            PersonKey("Joe", "West") to Person("Joe", "jwest@ccpd.com", 45000),
            PersonKey("Barry", "Allen") to Person("Barry", "ballen@starlabs.com", 88000)
        )

    @get:GetMapping("/byserializer")
    val allPersonsByPersonKeyWithJ: Map<PersonKeyWithJ, Person>?
        get() {
            map = mapOf(
                PersonKeyWithJ("Cisco", "Ramon") to Person("Cisco", "cramon@starlabs.com", 54500),
                PersonKeyWithJ("Iris", "West") to Person("Iris", "iwest@citizen.com", 78000)
            )
            return map
        }
}